core/services/payment_methods/forms/PayPalSettingsForm.php

```diff
@@ -0,0 +1,219 @@
+        parse_str(urldecode($response['body']), $response_args);
+
+        if (empty($response_args['ACK'])) {
+            EE_Error::add_error(
+                esc_html__(
+                    'Your PayPal credentials could not be verified. Part of their response was missing.',
+                    'event_espresso'
+                ),
+                __FILE__,
+                __FUNCTION__,
+                __LINE__
+            );
+        }
+        if (in_array(
+            $response_args['ACK'],
+            array(
+                'Success',
+                'SuccessWithWarning'
+            ),
+            true
+        )
+        ) {
+            return '';
+        } else {
+            return sprintf(
+                // translators: %1$s: PayPal response message, %2$s: PayPal response code
+                esc_html__(
+                    // @codingStandardsIgnoreStart
+                    'Your PayPal API credentials appear to be invalid. PayPal said "%1$s (%2$s)". Please see tips below.',
+                    // @codingStandardsIgnoreEnd
+                    'event_espresso'
+                ),
+                isset($response_args['L_LONGMESSAGE0'])
+                    ? $response_args['L_LONGMESSAGE0']
+                    : esc_html__('No error message received from PayPal', 'event_espresso'),
+                isset($response_args['L_ERRORCODE0']) ? $response_args['L_ERRORCODE0'] : 0
+            );
+        }
+    }
+
+    /**
+     * Gets the HTML to show the link to the help tab
+     * @return string
+     */
+    protected function helpTabLink()
+    {
+        return $this->helpTabLink;
+    }
+
+    /**
+     * Does the normal validation, but also verifies the PayPal API credentials work.
+     * If they don't, sets a validation error on the entire form, and adds validation errors (which are really more
+     * tips) on each of the inputs that could be the cause of the problem.
+     * @throws EE_Error
+     */
+    public function _validate()
+    {
+        parent::_validate();
+        $credentials_message = $this->checkForCredentialsErrors();
+        if ($credentials_message !== '') {
+            $this->add_validation_error($credentials_message);
+            $this->get_input('PMD_debug_mode')->add_validation_error(
+                esc_html__(
+                    // @codingStandardsIgnoreStart
+                    'If you are using PayPal Sandbox (test) credentials, Debug mode should be set to "Yes". Otherwise, if you are using live PayPal credentials, set this to "No".',
+                    // @codingStandardsIgnoreEnd
+                    'event_espresso'
+                )
+            );
+            $this->get_input('api_username')->add_validation_error(
+                sprintf(
+                    // translators: $1$s HTML for a link to the help tab
+                    esc_html__(
+                        'Are you sure this is your API username, not your login username?%1$s',
```

Comment:
maybe:

>  'Are you sure this is your API username and not your login username? %1$s',

Reply:
When I read that "and", it reads like "Are you sure this is your x and y?" (where x is "API username" and y is "not your login username"). Which reads a little strange.
What about "Are you sure this is your API username? (Not your login username)"?

Reply:
> it reads like "Are you sure this is your x and y?"

just throw the "not" in there and it becomes:

> "Are you sure this is your x and **not** y?"

or if you really want to think like a programmer:L

> "Are you sure this is your x and **!** y?"

or what about:

```
if ( $you === 'sure' && isApiCredential($username) && ! isLoginCredential($username) ) {
```